### PR TITLE
in_cpu/in_mem: add an option in config file to set the collection time interval

### PR DESCRIPTION
**Pull request by lorisbc**
Sometimes, it could be useful to set the collection time interval for the cpu
or memory measures. For example, if we want to store theses logs for estimate
the average of cpu consumption over a month, it is not useful to store mem 
every 1 second but maybe 60 seconds.
Maybe someone could also be interested in a very accurate cpu/mem 
measurement, and set collection time less than 1 second with nanoseconds.

By default, collect interval is 1 second and 0 nanosecond. 
To set a different interval, set in the .conf file: 
Collect_Sec 1
Collect_NSec 0

### CODE REVIEW

**Comment by edsiper** [comment]
thanks for the PR.

please rename the configuration option to Interval_Sec and Interval_NSec such as head input plugin does:

    http://fluentbit.io/documentation/0.11/input/head.html

Also please make sure to fix the indentation on plugins registration struct (already commented)

thanks

**Comment by edsiper** [comment]
thanks for the changes.

One more thing, would you please double check that when collecting snapshots metrics in less than a second (e.g sec=0 nsec=500000) get good results considering that we are using the CPU ticks to perform the calculation (CPU ticks are based in a second unit by the OS).

At the moment I cannot validate if that would be an issue or not, but if you can justify if is OK I will go ahead and merge this PR. Otherwise a workaround would be to just leave interval_sec and do not include the internal_nsec option for in_cpu to avoid bad metrics.

let me know your comments.

**Comment by lorisbc** [comment]
You are right, I stressed my CPU to test, and with less than 1 second, I have bad results. I will remove the interval_nsec option and just leave interval_sec.

Thanks

**Comment by lorisbc** [comment]
There is another issue with CPU even with second. If we set interval_sec to 5 seconds, the records will be the sum of 1 second records. For example, if the CPU is 100% during 20s, with a 5 seconds interval, we will have 4 records with 500% of usage instead of 100%. 
I will investigate on this issue.

File: plugins/in_cpu/in_cpu.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ static inline double CPU_METRIC_USAGE(unsigned long pre, unsigned long now,
     }
 
     diff = ULL_ABS(now, pre);
-    total = (diff * 100) / ctx->cpu_ticks;
+    total = (diff * 100) / ctx->cpu_ticks / ctx->interval_sec;
```
Comment (edsiper) [inline plugins/in_cpu/in_cpu.h]: Would you please add some parenthesis to protect cases where a division by zero could happen ?

File: plugins/in_cpu/in_cpu.h [hunk 2]
```diff
@@ -106,7 +106,13 @@ static inline double CPU_METRIC_SYS_AVERAGE(unsigned long pre, unsigned long now
     }
 
     diff = ULL_ABS(now, pre);
-    total = ((diff / ctx->cpu_ticks) * 100) / ctx->n_processors / ctx->interval_sec;
+
+    if (ctx->interval_sec > 0) {
+        total = ((diff / ctx->cpu_ticks) * 100) / ctx->n_processors / ctx->interval_sec;
```
Comment (edsiper) [inline plugins/in_cpu/in_cpu.h]: I mean:

```
  total = ((diff / ctx->cpu_ticks) * 100) / ctx->n_processors / ctx->interval_sec;
```

to 

```
  total = (((diff / ctx->cpu_ticks) * 100) / ctx->n_processors) / ctx->interval_sec;
```

**Comment by edsiper** [comment]
thanks!

**Pull request merged**